addons/binding/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/handler/HiveBridgeHandler.java

```diff
@@ -0,0 +1,357 @@
+                        .header("Accept", "application/vnd.alertme.zoo-6.1+json")
+                        .header("Content-Type", "application/vnd.alertme.zoo-6.1+json")
+                        .header("X-Omnia-Client", "Openhab 2").header("X-Omnia-Access-Token", token)
+                        .timeout(DISCOVER_TIMEOUT_SECONDS, TimeUnit.SECONDS).send();
+                int statusCode = response.getStatus();
+
+                if (statusCode == HttpStatus.UNAUTHORIZED_401) {
+                    // Token expired, get a new one and try again
+                    getToken();
+                    statusCode = response.getStatus();
+                }
+
+                if (statusCode != HttpStatus.OK_200) {
+                    String statusLine = response.getStatus() + " " + response.getReason();
+                    logger.error("Method failed: {}", statusLine);
+                    return;
+                }
+
+                Gson gson = new GsonBuilder().create();
+                HiveNodes o = gson.fromJson(response.getContentAsString(), HiveNodes.class);
+
+                if (o.nodes.size() > 0) {
+                    // Loop through the nodes and add thermostats but only if they are showing a temperature as the api
+                    // reports lots of thermostats for some reason
+                    HiveNode thermostat = null;
+                    HiveNode systemInfo = null;
+                    for (HiveNode node : o.nodes) {
+                        if (node.attributes != null && node.attributes.nodeType != null
+                                && node.attributes.nodeType.reportedValue.equals(THERMOSTAT_NODE_TYPE)
+                                && node.attributes.temperature != null) {
+                            // If this a thermostat, and it has a current temperature, add it to discovery results
+                            thermostat = node;
+                        }
+                        if (node.attributes != null && node.attributes.batteryLevel != null) {
+                            // If this a thermostat, and it has a current temperature, add it to discovery results
+                            systemInfo = node;
+                        }
+                    }
+                    if (thermostat != null) {
+                        if (systemInfo != null) {
+                            thermostat.linkedNode = systemInfo.id;
+                            thermostat.firmwareVersion = systemInfo.attributes.softwareVersion.displayValue;
+                            thermostat.model = systemInfo.attributes.model.displayValue;
+                            thermostat.macAddress = systemInfo.attributes.macAddress.displayValue;
+                        }
+                        hiveDiscoveryService.addDevice(thermostat);
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("Failed to add node to discovery results: {}", e);
```

Comment:
1. don't use the error log level that is reserved to the framework
2. don't catch generic `Exception`s. Keep the catch block as small as possible. It is rather large here.
3. Add the exception as last, unbound (no "{}") parameter to a logger statement for printing the stacktrace. If you don't want the stacktrace (for errors that happen frequently and controlled) you can use `e.getMessage()`.